docs/eventing/sources/pingsource.md

```diff
@@ -1,196 +1,261 @@
+This topic explains event-driven architecture in terms of producers and consumers and, to that end,
+shows how to configure PingSource as an event source targeting a Knative Service.
 
-A PingSource produces events with a fixed payload on a specified cron schedule.
+## Before you begin
 
-## Installation
+1. Set up [Knative Serving](../../../serving).
+1. Set up [Knative Eventing](../../../eventing).
 
-The PingSource source type is enabled by default when you install Knative Eventing.
+## Overview
 
-## Example
+In event-driven architecture the most straightforward use case is a producer creating an event and a
+consumer acting on that event.
 
-This example shows how to send an event every minute to a Event Display Service.
+### Producers
 
-### Creating a namespace
+Producers create events. Using GitHub as an example, the producer is the GitHub webhook that sends
+information when a pull request is created.
+Essentially, you ask the producer for events that they are capable of sending and you tell the
+producer where to send those events.
 
-Create a new namespace called `pingsource-example` by entering the following
-command:
+Producers send the event data per specifications that make it easy for consumers to handle it.
+Knative uses the CloudEvents specifications by default.
+For more information, see the [Cloud Events](https://cloudevents.io/) website.
 
-```shell
-kubectl create namespace pingsource-example
-```
+### Consumers
+
+Consumers consume the data that producers create. Consumers are usually functions.
 
-### Creating the Event Display Service
+In the GitHub example above, a consumer might update a ticket in the ticketing system with the data
+from the pull request.
 
-In this step, you create one event consumer, `event-display` to verify that
-`PingSource` is properly working.
 
-To deploy the `event-display` consumer to your cluster, run the following
-command:
+## Create a consumer (Knative service)
+
+To create a consumer, and verify that `PingSource` is working, create a simple Knative service that
+receives CloudEvents and writes them to stdout.
+
+{{< tabs name="create-service" default="YAML" >}}
+{{% tab name="YAML" %}}
+Create the service from stdin by running:
 
 ```shell
-kubectl -n pingsource-example apply -f - << EOF
-apiVersion: apps/v1
-kind: Deployment
+cat <<EOF | kubectl create -f -
+apiVersion: serving.knative.dev/v1
+kind: Service
 metadata:
   name: event-display
 spec:
-  replicas: 1
-  selector:
-    matchLabels: &labels
-      app: event-display
   template:
-    metadata:
-      labels: *labels
     spec:
       containers:
-        - name: event-display
-          image: gcr.io/knative-releases/knative.dev/eventing-contrib/cmd/event_display
+        - image: gcr.io/knative-releases/knative.dev/eventing-contrib/cmd/event_display
+EOF
+```
+{{< /tab >}}
 
----
+{{% tab name="kn" %}}
+Create the service using the kn command-line interface by running:
```

Comment:
```suggestion
Create a service using the `kn` CLI:
```
CLI is a common enough term to abbreviate IMO, we use CLI most other places I believe.

Reply:
Google Style says to spell it out: https://developers.google.com/style/word-list#letter-c
We can over-rule, of course, but if so that needs to go in our Word & Phrase list first.

Reply:
Added to the list, let's overrule it please, I don't think this is an uncommon enough phrase for us to need to spell it out every time.